src/torchcodec/_core/Encoder.cpp

```diff
@@ -228,7 +233,7 @@ void AudioEncoder::encode() {
   avFrame->format = AV_SAMPLE_FMT_FLTP;
   avFrame->sample_rate = avCodecContext_->sample_rate;
   avFrame->pts = 0;
-  setChannelLayout(avFrame, avCodecContext_);
+  setDefaultChannelLayout(avFrame, static_cast<int>(wf_.sizes()[0]));
```

Comment:
I had to think about this for a few moments to convince myself it's correct, so it may be worth putting in a comment: the _default_ channel layout should be the channel layout of the provided waveform.